pkg/k8shandler/common_test.go

```diff
@@ -542,63 +543,17 @@ func buildNoCPULimitResource(cpuRequest, memLimit, memRequest resource.Quantity)
 }
 
 func areResourcesSame(lhs, rhs v1.ResourceRequirements) bool {
-
-	if !areQuantitiesSame(lhs.Limits.Cpu(), rhs.Limits.Cpu()) {
-		return false
-	}
-
-	if !areQuantitiesSame(lhs.Requests.Cpu(), rhs.Requests.Cpu()) {
-		return false
-	}
-
-	if !areQuantitiesSame(lhs.Limits.Memory(), rhs.Limits.Memory()) {
-		return false
-	}
-
-	if !areQuantitiesSame(lhs.Requests.Memory(), rhs.Requests.Memory()) {
-		return false
-	}
-
-	return true
+	return printResource(lhs) == printResource(rhs)
```

Comment:
in general, you are *not* guaranteed the ordering of elements.  Is the ordering guaranteed in this case?  If so, is there a citation?  I don't want to make this code more fragile.

Reply:
The issue occurs because we are directly comparing stucts and not the value of what is ultimately serialized and ingested by the apiserver.  Though we are not guaranteed of the order, I think the struct is reasonably small and serialization is generally  consistent .  I don't believe we see cases where for instance the `meta` sometimes is serialized after `spec`

Reply:
It seems backwards that comparing the actual data doesn't work, and we have to serialize to string to do the actual comparison.  Maybe structs work differently than maps/arrays - with maps/arrays you can use `reflect.DeepEqual`

Reply:
let me investigate. Maybe it was because of the way we displayed failure in the print

Reply:
I only am able to demonstrate the difference when the struct is serialized.  Unfortunately 0 and nil in this context are not considered to be equivalent

Reply:
@bparees what's the best way to compare structs like this?

Reply:
quantities where you want zero and nil to be treated as equivalent?  i'm not sure we have a generic utility for that.

Reply:
@bparees then is marshal to string, and compare strings, ok?

@jcantrill it really seems as if you need to write a custom comparison for `Resource` objects.  I know you said ". . .not the value of what is ultimately serialized and ingested by the apiserver." - Yes, the api server is going to receive a serialized string - but the api server is going to unmarshal that into objects and use the objects, not the serialized string.

Reply:
why can't we just fix the comparison to handle nil/zero explicitly?

Reply:
@bparees im not sure I understand what you mean?  This does an equivalent comparison.  @richm the apiserver is going to unmarshal into an object but `ResourceLimits` is a type alias to a maps so no entry means you have nothing for cpu/or memory versus zero.  It's different.

Reply:
i mean "if (lhs.Limits.CPU==0 || lhs.Limts.CPU==nil) && (rhs.Limits.CPU==0 || lhs.Limits.CPU==nil) then they are equal"

Reply:
`reflect.DeepEquals` exposes the issue properly